### PR TITLE
Update TensorRT img2img community pipeline

### PR DESCRIPTION
**Pull request by asfiyab-nvidia**
# What does this PR do?

1. Refactor TensorRT Img2Img pipeline to inherit from `DiffusionPipeline` 
2. Update TensorRT API to align with the latest version (10.2.0)

## Before submitting
- [ x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ y] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ y] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ y] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ x] Did you write any new necessary tests?

## Who can review?

cc @yiyixuxu @asomoza

### CODE REVIEW

**Comment by tolgacangoz** [comment]
Thanks for the refactor!
[Could you make style and quality proper :)](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#how-to-open-a-pr:~:text=%F0%9F%A7%A8%20Diffusers%20relies%20on,%24%20make%20quality)

**Comment by asfiyab-nvidia** [comment]
@tolgacangoz thanks for the link. Updated the style and quality in a follow up commit

**Comment by tolgacangoz** [comment]
Running `make style && make quality` for full formatting is necessary.

**Comment by tolgacangoz** [comment]
Running `make style && make quality` for full formatting is necessary.
Thanks again for the refactor!

**Comment by tolgacangoz** [comment]
Running `make style && make quality` for full formatting is necessary.
Thanks again for the refactor!

File: examples/community/README.md [hunk 1]
```diff
@@ -1641,18 +1641,18 @@ from io import BytesIO
 from PIL import Image
 import torch
 from diffusers import DDIMScheduler
-from diffusers.pipelines.stable_diffusion import StableDiffusionImg2ImgPipeline
+from diffusers.pipelines import DiffusionPipeline
```
Comment (tolgacangoz) [inline examples/community/README.md:1644]:
```suggestion
from diffusers import DiffusionPipeline
```

File: examples/community/stable_diffusion_tensorrt_img2img.py [hunk 2]
```diff
@@ -41,12 +42,13 @@
     network_from_onnx_path,
     save_engine,
 )
-from polygraphy.backend.trt import util as trt_util
 from transformers import CLIPFeatureExtractor, CLIPTextModel, CLIPTokenizer, CLIPVisionModelWithProjection
 
+from diffusers.configuration_utils import FrozenDict, deprecate
+from diffusers.image_processor import VaeImageProcessor
 from diffusers.models import AutoencoderKL, UNet2DConditionModel
+from diffusers.pipelines.pipeline_utils import DiffusionPipeline
```
Comment (tolgacangoz) [inline examples/community/stable_diffusion_tensorrt_img2img.py:50]:
```suggestion
from diffusers import DiffusionPipeline
```

**Review by tolgacangoz — APPROVED** [review]
Thanks for the refactor! I just suggested simplifying. Also, running `make style && make quality` for full formatting is necessary.

**Comment by asfiyab-nvidia** [comment]
> Thanks for the refactor! I just suggested simplifying. Also, running `make style && make quality` for full formatting is necessary.

Thanks for the suggested simplifications - committed them. 
Running `make style && make quality` resulted in the changes in [commit](https://github.com/huggingface/diffusers/pull/8899/commits/442ab2dc7704562f41f90cf5bc8d9dbbd37a121d). Is there more I'm missing?

**Comment by tolgacangoz** [comment]
I guess your `ruff`'s version was not 0.1.5. Could you run `make style && make quality` with this `ruff` version after revoking the last commit?

**Comment by asfiyab-nvidia** [comment]
> Is your `ruff`'s version [0.1.5](https://github.com/huggingface/diffusers/blob/f57b27d2ad52b7d3958a83d9ec4ce4cb6d42f615/setup.py#L124)? Could you run `make style && make quality` after revoking the last commit and `python -m pip install -e ".[dev]"`?

Updated using ruff 0.1.5 in the latest commit. Hope this passes the style checker

**Comment by tolgacangoz** [comment]
LGTM; thanks!

**Comment by asfiyab-nvidia** [comment]
@DN6 @tolgacangoz Can I get help moving this PR along? I looked at both failures and they seem unrelated to community pipeline change I made

**Pull request merged**